src/kudu/gutil/stl_util.h

```diff
@@ -839,7 +839,7 @@ class STLCountingAllocator : public Alloc {
         bytes_used_(x.bytes_used()) {
   }
 
-  pointer allocate(size_type n, std::allocator<void>::const_pointer hint = 0) {
+  pointer allocate(size_type n, std::allocator_traits<std::allocator<void>>::const_pointer hint = 0) {
```

Comment:
std::allocator has changed in C++20. The Allocator class spec is here: https://en.cppreference.com/w/cpp/named_req/Allocator

A number of the properties and functions are specified to be optional, like `const_pointer` and `max_size()`. The idea is that if they don't exist, it is assumed that they follow the defaults.

In C++11 and below, `std::allocator` just implemented them anyway but explicitly returned the default values. In C++17, it was deprecated, and the correct way is to use `std::allocator_traits` to evaluate these (introduced since C++11). `std::allocator_traits` checks if the underlying allocator implemented the method/property. If implemented, the implementation is used, if not the default is returned.

In C++20, the default property/methods are removed from `std::allocator`, so our old code here doesn't compile anymore. The default values returned by them lives in `std::allocator_traits` however, so we can use that instead.